src/import/ImportWavPack.cpp

```diff
@@ -228,20 +233,26 @@ ProgressResult WavPackImportFileHandle::Import(WaveTrackFactory *trackFactory, T
          }
 
          totalSamplesRead += samplesRead;
-         updateResult = mProgress->Update(WavpackGetProgress(mWavPackContext), 1.0);
-      } while (updateResult == ProgressResult::Success && samplesRead != 0);
+         
+         progressListener.OnImportProgress(WavpackGetProgress(mWavPackContext));
+      } while (!mCancelled && samplesRead != 0);
    }
 
    if (WavpackGetNumErrors(mWavPackContext))
       AudacityMessageBox( XO( "Encountered %d errors decoding WavPack file!" ).Format( WavpackGetNumErrors(mWavPackContext) ),
                           XO( "WavPack Importer" ), wxOK | wxICON_EXCLAMATION | wxCENTRE);
 
-   if (updateResult != ProgressResult::Stopped && updateResult != ProgressResult::Cancelled
-         && totalSamplesRead < mNumSamples)
-      updateResult = ProgressResult::Failed;
-
-   if (updateResult == ProgressResult::Failed || updateResult == ProgressResult::Cancelled)
-      return updateResult;
```

Comment:
In the old logic, there was a distinction between stopping and cancelling.  In case of stopping, there was no error reported, even if `totalSamplesRead < mNumSamples`.

In the new logic, this changes.

Reply:
Progress dialogs had a Stop button distinct from Cancel in the previous version, but with this commit they don't.

Is this change intended?

Reply:
I think that the major distinction between them is that the data that has been been read by the time Stop button was pressed is preserved, but mpg123 importer seem to be an exception. Will update this commit so that this behavior is preserved.